lib/puppet/provider/iam_instance_profile/v2.rb

```diff
@@ -0,0 +1,92 @@
+require_relative '../../../puppet_x/puppetlabs/aws.rb'
+
+require 'uri'
+
+Puppet::Type.type(:iam_instance_profile).provide(:v2, :parent => PuppetX::Puppetlabs::Aws) do
+  confine feature: :aws
+
+  mk_resource_methods
+
+  read_only(:arn)
+
+  def self.instances
+    response = iam_client.list_instance_profiles()
+    response.instance_profiles.collect do |instance_profile|
+      new({
+              name: instance_profile.instance_profile_name,
+              ensure: :present,
+              path: instance_profile.path,
+              arn: instance_profile.arn,
+              roles: instance_profile.roles,
+          })
+    end
+  end
+
+  def self.prefetch(resources)
+    instances.each do |prov|
+      if resource = resources[prov.name]
+        resource.provider = prov
+      end
+    end
+  end
+
+  def exists?
+    Puppet.info("Checking if IAM instance profile #{name} exists")
```

Comment:
I've replaced all the rest of these to be `debug`.  Would you mind updating these messages to use `debug` also please?